kratos/utilities/intersection_utilities.h

```diff
@@ -232,6 +232,56 @@ class IntersectionUtilities
         return 0;
     }
 
+    /**
```

Comment:
A unit test must be added for this function in `test_intersection_utilities.cpp`.